src/alire/alire-toolchains-solutions.adb

```diff
@@ -1,17 +1,143 @@
+with AAA.Strings;
+
+with Alire.Config.Edit;
+with Alire.Directories;
+with Alire.Index;
+with Alire.Paths;
+with Alire.Platforms.Folders;
 with Alire.Root;
 with Alire.Shared;
 
+with CLIC.User_Input;
+
 package body Alire.Toolchains.Solutions is
 
+   -------------------
+   -- Migrate_Cache --
+   -------------------
+
+   procedure Migrate_Cache is
```

Comment:
I don't like having such a big piece of code for something that only has to be done once. Downloading dependencies is not that bad of an issue.